### PR TITLE
CONTRIBUTING.md: add instructions for testing PRs

### PR DESCRIPTION
**Pull request by classabbyamp**
I've had to find these instructions time and time again for requesting testing from people, so I figured why not make it part of our documentation. Maybe it will help get more users involved with testing.

Open to suggestions on wording, grammar, missing info, etc.

### CODE REVIEW

**Comment by tranzystorek-io** [comment]
Do we want to suggest specific things to look at when testing? E.g., checking manpages, help prompts, runits?

**Comment by Duncaen** [comment]
> Do we want to suggest specific things to look at when testing? E.g., checking manpages, help prompts, completion, runits?

I don't think that would be necessary, we are more interested in users of the software actually testing if their setup works, not random people going trough a superficial checklist, which the contributor would probably already have done.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -129,3 +129,24 @@ Please abstain from temporarily closing a pull request while revising the templa
 
 Once the reviewers have merged the pull request, our [build server](http://build.voidlinux.org) is automatically triggered and builds
 all packages in the pull request for all supported platforms. Upon completion, the packages are available to all Void Linux users.
+
+## Testing Pull Requests
+
+Testing new package submissions and updates is always helpful, and is a great way to get started with contributing.
```
Comment (Chocimier) [inline CONTRIBUTING.md:135]: I would introduce it with something like

> While it is PR creator responsibility to test changes before sending it, one person won't test all configuration options, usecases, hardware etc.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -129,3 +129,24 @@ Please abstain from temporarily closing a pull request while revising the templa
...
+First, [clone the repository](https://github.com/void-linux/void-packages#quick-start) if you haven't done so already.
+Then check out the pull request, either with `github-cli`:
+
+    $ gh pr checkout <number>
+
+Or manually:
```
Comment (Chocimier) [inline CONTRIBUTING.md:141]: Or with `git`.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -129,3 +129,24 @@ Please abstain from temporarily closing a pull request while revising the templa
...
+
+    $ git fetch origin pull/<number>/head:<branch-name>
+    $ git checkout <branch-name>
+
+If your local void-packages repository is cloned from your fork, you may need to add the main repository as a remote first:
+
+    $ git remote add upstream https://github.com/void-linux/void-packages.git
```
Comment (Chocimier) [inline CONTRIBUTING.md:148]: This should go before fetching to preserve time order, and before `gh` if applies there.

Comment (classabbyamp) [reply CONTRIBUTING.md:146]: tested and `gh` is able to do it without adding any remotes from either the fork clone or base repo clone

**Review by Chocimier — APPROVED** [review]

**Pull request merged**